### ISSUE TITLE
[Bug]: Get different results on different platforms

### ISSUE THREAD

**Issue by ouyuanning** [matrixorigin/matrixone#3821]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

select cast(19999999999999999999 as signed);

in Mac(Arm)，  we get 
9223372036854775807

in CI Env (Linux(x86_64)) ，  we get
-9223372036854775808

**Comment by ouyuanning** [comment]
we fix #3785  then get #3821

**Comment by aressu1985** [comment]
x86:
mysql> select cast(19999999999999999999 as signed);
ERROR 20201 (HY000): data out of range: data type int64, value '19999999999999999999'

mac:
mysql> select cast(19999999999999999999 as signed);
ERROR 20201 (HY000): data out of range: data type int64, value '19999999999999999999'
